In [2]:
from math import factorial

**Задача №1**.  Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

Вероятность того, что стрелок попадет в цель ровно 85 раз (событие А) определим по формуле Бернулли: <br>
 $\displaystyle P_{n}(X=k)=C^k_{n}p^{k}q^{n-k}$, где: <br>
n - число испытаний, <br>
k - число наступления события, <br>
p - вероятность наступления события A, <br>
q - противоположная вероятность q = 1 – p <br>



Определим какое количество вариантов (общее число исходов) есть, чтобы стрелок из 100 выстрелов попал в мишень ровно 85 раз по следующей формуле комбинаторики (сочетание): <br>
$\displaystyle C^k_n=\frac{n!}{k!(n-k)!}$

In [18]:
def hit_the_target (n, k):
  return int(factorial(n) / (factorial(k) * factorial(n - k)))


In [19]:
hit_the_target (100, 85)

253338471349988640

Вероятность того, что стрелок попадет в цель ровно 85 раз (событие А):

In [20]:
def probability (p, n, k):
    return hit_the_target (n,k) * p**k * (1-p)**(n-k)

In [21]:
round(probability (0.8, 100, 85), 4)

0.0481

**Ответ:** Итого вероятность того, что стрелок попадет в цель ровно 85 раз (событие А) составляет: <br>
 Р(А) = 0,0481

**Задача №2.** Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?

Вероятность того, что лампочка перегорит исходя из заданных условий определим по формуле Пуассона: <br>
 $\displaystyle P_m=\frac{\lambda^m}{m!}\cdot e^{-\lambda}\text{, где: }\lambda \text{ среднее количество наступления события за определенную единицу измерений, и }\lambda=p\cdot n$

In [37]:
import numpy as np

Вероятность того, что лампочка перегорит исходя из заданных условий определим следующим образом:

In [40]:
def lamp_burned_out (m, n, p):
    l = p * n
    return l**m/factorial(m) * np.exp(-l)

Вероятность, что ни одна из лампочек не перегорит в первый день:

In [41]:
p0 = round(lamp_burned_out(0, 5000, 0.0004), 4)
p0

0.1353

Вероятность, что перегорят ровно две лампочки в первый день:

In [42]:
p1 = round(lamp_burned_out(2, 5000, 0.0004), 4)
p1

0.2707

**Ответ:** Вероятность, что ни одна из лампочек не перегорит в первый день: 0,1353. <br> Вероятность, что перегорят ровно две лампочки в первый день: 0,2707.

**Задача №3.** Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

Вероятность того, что орел выпадет ровно 70 раз из 144 подбрасываний (событие А) определим по формуле Бернулли: <br>
 $\displaystyle P_{n}(X=k)=C^k_{n}p^{k}q^{n-k}$, где: <br>
n - число испытаний, <br>
k - число наступления события, <br>
p - вероятность наступления события A, <br>
q - противоположная вероятность q = 1 – p <br>

Определим какое количество вариантов (общее число исходов) есть, чтобы орел выпал 70 раз из 144 раз по следующей формуле комбинаторики (сочетание): <br>
$\displaystyle C^k_n=\frac{n!}{k!(n-k)!}$

In [22]:
def eagle (n, k):
  return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [23]:
eagle (144, 70)

1400749509083708712353342007350408066891776

Вероятность того, что орел выпадет ровно 70 раз из 144 (событие А):

In [26]:
def probabilityEagle (p, n, k):
    return eagle (n,k) * p**k * (1-p)**(n-k)

In [27]:
round(probabilityEagle (0.5, 144, 70), 4)

0.0628

**Ответ:** Итого вероятность того, что орел выпадет ровно 70 раз из 144 подбрасываний (событие А) составляет: <br>
 Р(А) = 0,0628.

**Задача №4.** В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. <br>
a) Какова вероятность того, что все мячи белые? <br>
б) Какова вероятность того, что ровно два мяча белые? <br>
в) Какова вероятность того, что хотя бы один мяч белый?

**а)** Определим какое количество вариантов (общее число исходов) есть, чтобы при вытаскивании из ящиков по два мяча, все мячи оказались белыми по следующей формуле комбинаторики (сочетание): <br>
$\displaystyle C^k_n=\frac{n!}{k!(n-k)!}$

In [3]:
def ballBox1 (n, k):
  return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [4]:
 ballBox1 (10, 2)

45

In [5]:
n1=45

In [6]:
def ballBox2 (n, k):
  return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [7]:
ballBox2 (11, 2)

55

In [8]:
n2=55

Определим какое количество вариантов (благоприятных исходов) есть, чтобы при вытаскивании из ящиков по два мяча, все мячи оказались белыми по следующей формуле комбинаторики (сочетание): <br>
$\displaystyle C^k_n=\frac{n!}{k!(n-k)!}$

In [9]:
def ballBox1good (n, k):
  return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [10]:
ballBox1 (7, 2)

21

In [11]:
m1=21

In [12]:
def ballBox2good (n, k):
  return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [13]:
ballBox2good (9, 2)

36

In [14]:
m2=36

Вероятность одновременного наступления событий А и В, а именно, что извлеченные из 2х ящиков по два мяча будут белые  определим по формуле: <br>
$\displaystyle P(A+В)=P(A)\cdot P(B)$ <br>

$\displaystyle P(A)=\frac{m}{n}$, <br>

$\displaystyle P(В)=\frac{m}{n}$,
 <br> где $m$ - число благоприятных исходов (наступления события А), <br> $n$ - общее число всех исходов.

In [18]:
PA=round(m1/n1, 4)

In [19]:
PA

0.4667

In [20]:
PB=round(m2/n2, 4)

In [21]:
PB

0.6545

In [24]:
PAB=round(PA*PB, 4)

In [25]:
PAB

0.3055

**Ответ:** Вероятность одновременного наступления событий А и В, а именно, что извлеченные из 2х ящиков по два мяча будут белые составляет P(A)=0.3055

**б)** Определим какое количество вариантов есть, чтобы при вытаскивании из ящиков по два мяча, только два мяча оказались белыми по следующей формуле комбинаторики (сочетание): <br>
$\displaystyle C^k_n=\frac{n!}{k!(n-k)!}$

In [3]:
def ballsArguments(n, k):
  return int(factorial(n) / (factorial(k) * factorial(n - k)))

Вероятность события А определим по формуле: <br>
P(A)=m/n, где: <br>   m  - число благоприятных исходов (наступления события А), <br>
n  - общее число всех исходов.

In [5]:
goodTwo  = ballsArguments(7,2)*ballsArguments(2,2) + \
ballsArguments(7,1)*ballsArguments(3,1)*ballsArguments(9,1)*ballsArguments(2,1) + \
ballsArguments(9,2)*ballsArguments(3,2)
goodAll = ballsArguments(10,2) * ballsArguments(11,2)
twoWhiteBalls = round(goodTwo/goodAll, 4)
twoWhiteBalls

0.2048

**Ответ:** Вероятность того, при вытаскивании из ящиков по два мяча, только два мяча оказались белыми составляет P(A)=0.2048

**в)** Определим какое количество вариантов есть, чтобы при вытаскивании из ящиков по два мяча, только один мяч оказался белым по следующей формуле комбинаторики (сочетание): <br>
$\displaystyle C^k_n=\frac{n!}{k!(n-k)!}$

In [7]:
def ballsArguments(n, k):
  return int(factorial(n) / (factorial(k) * factorial(n - k)))

Вероятность события $\bar A$-"не было извлечено ни одного белого мяча", которое наряду с искомым событием $A$-"был извлечен хотя бы 1 белый мяч", составляют полную группу событий, по скольку в результате испытаний может произойти только либо событие $A$, либо событие $\bar A$, и поэтому:  
$P(A)=1-P(\bar A)$, где вероятность событий определим по формуле: <br>
P=m/n, где: <br>   m  - число благоприятных исходов, <br>
n  - общее число всех исходов.<br>
Для наступления события $\bar A$ необходимо чтобы из первой и второй корзин не было вынуто ни одного белого мяча, число таких благоприятных исходов для первой корзины $m_1=C^2_3$ а для второй $C^2_2$  
Вычислим $m_1$ и $m_2$ (число благоприятных исходов при извлечение мячей из ящиков 1 и 2:

In [9]:
m1=ballsArguments(3,2)

In [19]:
m1

3

In [12]:
m2=ballsArguments(2,2)

In [20]:
m2

1

Вычислим $n_1$ и $n_2$ (общее число всех исходов при извлечение мячей из ящиков 1 и 2):

In [23]:
n1=ballsArguments(10,2)

In [26]:
n1

45

In [25]:
n2=ballsArguments(11,2)

In [27]:
n2

55

Тогда $P(\bar A)=\frac{m1}{n1}\cdot\frac{m2}{n2}$, вычислим $P(\bar A)$ и $P(A)$:

In [35]:
PAreverse = round((m1/n1)*(m2/n2), 4)

In [36]:
PAreverse

0.0012

In [37]:
PA=1-PAreverse

In [38]:
PA

0.9988

**Ответ:** Вероятность того, что при вытаскивании из ящиков по два мяча, только один мяч окажется белым составляет P(A)=0.9988